### PR TITLE
[FLINK-35430][cdc-connector][kafka] Passed the zoneId to JsonSerializationSchema.

### PR DESCRIPTION
**Pull request by joyCurry30**
Refer to [FLINK-35430](https://issues.apache.org/jira/browse/FLINK-35430)

Passed the zoneId to JsonSerializationSchema.

### CODE REVIEW

**Review by lvyanquan — APPROVED** [review]
LGTM.

**Review by czy006 — APPROVED** [review]
LGTM

**Comment by yuxiqian** [comment]
Hi @joyCurry30, could you please create another PR to backport this to `release-3.1`? We're preparing for the next patch release, and it would be nice to include this bugfix.

**Comment by joyCurry30** [comment]
> Hi @joyCurry30, could you please create another PR to backport this to `release-3.1`? We're preparing for the next patch release, and it would be nice to include this bugfix.

https://github.com/apache/flink-cdc/pull/3377
Hi, cc plz. @yuxiqian

**Review by leonardBang — COMMENTED** [review]
Thanks @joyCurry30 for the fix and @lvyanquan @czy006 for the review, LGTM.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/json/canal/CanalJsonSerializationSchema.java [hunk 1]
```diff
@@ -81,12 +81,13 @@ public CanalJsonSerializationSchema(
             TimestampFormat timestampFormat,
             JsonFormatOptions.MapNullKeyMode mapNullKeyMode,
             String mapNullKeyLiteral,
+            ZoneId zoneId,
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/json/canal/CanalJsonSerializationSchema.java:84]: This should be okay for compatibility consideration as this class is not a public API

**Review by leonardBang — APPROVED** [review]

**Pull request merged**